vendor/bat-native-ads/src/bat/ads/internal/user_activity/user_activity_unittest.cc

```diff
@@ -0,0 +1,177 @@
+
+  // Act
+  UserActivity::Get()->RecordEvent(event_type);
+
+  // Assert
+  const UserActivityEventHistoryMap history =
+      UserActivity::Get()->get_history();
+  EXPECT_EQ(1UL, history.size());
+  EXPECT_NE(history.end(), history.find(event_type));
+}
+
+TEST_F(BatAdsUserActivityTest,
+    RecordPlayedMediaEvent) {
+  // Arrange
+  const UserActivityEventType event_type = UserActivityEventType::kPlayedMedia;
+
+  // Act
+  UserActivity::Get()->RecordEvent(event_type);
+
+  // Assert
+  const UserActivityEventHistoryMap history =
+      UserActivity::Get()->get_history();
+  EXPECT_EQ(1UL, history.size());
+  EXPECT_NE(history.end(), history.find(event_type));
+}
+
+TEST_F(BatAdsUserActivityTest,
+    RecordBrowserWindowDidBecomeActiveEvent) {
+  // Arrange
+  const UserActivityEventType event_type =
+      UserActivityEventType::kBrowserWindowDidBecomeActive;
+
+  // Act
+  UserActivity::Get()->RecordEvent(event_type);
+
+  // Assert
+  const UserActivityEventHistoryMap history =
+      UserActivity::Get()->get_history();
+  EXPECT_EQ(1UL, history.size());
+  EXPECT_NE(history.end(), history.find(event_type));
+}
+
+TEST_F(BatAdsUserActivityTest,
+    RecordBrowserWindowDidEnterBackgroundEvent) {
+  // Arrange
+  const UserActivityEventType event_type =
+      UserActivityEventType::kBrowserWindowDidEnterBackground;
+
+  // Act
+  UserActivity::Get()->RecordEvent(event_type);
+
+  // Assert
+  const UserActivityEventHistoryMap history =
+      UserActivity::Get()->get_history();
+  EXPECT_EQ(1UL, history.size());
+  EXPECT_NE(history.end(), history.find(event_type));
+}
+
+TEST_F(BatAdsUserActivityTest,
+    RecordTheSameEventMultipleTimes) {
+  // Arrange
+  const UserActivityEventType event_type =
+      UserActivityEventType::kOpenedNewOrFocusedOnExistingTab;
+
+  // Act
+  UserActivity::Get()->RecordEvent(event_type);
+  UserActivity::Get()->RecordEvent(event_type);
+
+  // Assert
+  const UserActivityEventHistoryMap history =
+      UserActivity::Get()->get_history();
+
+  const UserActivityEventHistory user_activity_event_history =
+      history.find(event_type)->second;
+
+  EXPECT_EQ(2UL, user_activity_event_history.size());
+}
+
+TEST_F(BatAdsUserActivityTest,
+    GetHistory) {
+  // Arrange
+  UserActivity::Get()->RecordEvent(
+      UserActivityEventType::kOpenedNewOrFocusedOnExistingTab);
+  UserActivity::Get()->RecordEvent(
+      UserActivityEventType::kOpenedNewOrFocusedOnExistingTab);
+  UserActivity::Get()->RecordEvent(
+      UserActivityEventType::kClosedTab);
+  UserActivity::Get()->RecordEvent(
+      UserActivityEventType::kPlayedMedia);
+  UserActivity::Get()->RecordEvent(
+      UserActivityEventType::kBrowserWindowDidBecomeActive);
+  UserActivity::Get()->RecordEvent(
+      UserActivityEventType::kBrowserWindowDidEnterBackground);
+
+  // Act
+  const UserActivityEventHistoryMap history =
+      UserActivity::Get()->get_history();
+
+  // Assert
+  EXPECT_EQ(5UL, history.size());
```

Comment:
@tmancey shouldn't that equal 6 given that 6 events were recorded? `RecordEvent` only seems to filter out if max entries are exceeded.

Reply:
5 is correct because we are only testing the key and not the values. Each activity has its own key

Reply:
ah ofc 👍